Doc: Bluetooth: Mesh: Added shell remote provisioning documentation (#55784)
zephyrproject-rtos/zephyr: doc/connectivity/bluetooth/api/mesh/shell.rst

	Abort the Opcodes Aggregator Sequence message. This command clears the Opcodes Aggregator Client context.

Remote Provisioning Client
--------------------------

> The Remote Provisioning client is an optional Bluetooth mesh model that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_RPR_CLI` configuration option. The Remote Provisioning Client model is used to support the functionality of dispatching a sequence of access layer messages to nodes supporting the Remote Provisioning Server model.

Reviewer: ```suggestion
The Remote Provisioning Client is an optional Bluetooth mesh model that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_RPR_CLI` configuration option. The Remote Provisioning Client model supports the functionality of dispatching a sequence of access layer messages to nodes supporting the Remote Provisioning Server model.
```

``mesh models rpr scan <timeout in seconds> [<UUID>]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	Start scanning for unprovisioned devices.

	* ``timeout in seconds``: Scan timeout in seconds. Must be at least 1 second.
	* ``UUID``: Device UUID to scan for, or NULL to report all devices.

``mesh models rpr scan-ext <timeout in seconds> <UUID> [<AD-type> ... ]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Start extended scanning for unprovisioned devices.

Reviewer: ```suggestion
	Start the extended scanning for unprovisioned devices.
```

> 	* ``timeout in seconds``: Scan timeout in seconds. Valid values from @ref BT_MESH_RPR_EXT_SCAN_TIME_MIN to @ref BT_MESH_RPR_EXT_SCAN_TIME_MAX.

Reviewer: Is there a reason for having @ref before the two values?
Are these config options, could it be written :kconfig:option: instead?
If not, enclose with ``.

Author: These are not kconfig options.

	* ``UUID``: Device UUID to start extended scanning for.
	* ``AD-type``: List of AD types to include in the scan report. Must contain 1 to :kconfig:option:`CONFIG_BT_MESH_RPR_AD_TYPES_MAX` entries.

``mesh models rpr scan-srv [<AD-type> ... ]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Start extended scanning for the remote server.

Reviewer: Could we say Remote Provisioning Server here as well?
```suggestion
	Start the extended scanning for the Remote Provisioning Server.
```

	* ``AD-type``: List of AD types to include in the scan report. Must contain 1 to :kconfig:option:`CONFIG_BT_MESH_RPR_AD_TYPES_MAX` entries.

``mesh models rpr scan-caps``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Get scanning capabilities of Remote Provisioning Server.

Reviewer: ```suggestion
	Get the scanning capabilities of the Remote Provisioning Server.
```

``mesh models rpr scan-get``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Get current scanning state of Remote Provisioning Server.

Reviewer: ```suggestion
	Get the current scanning state of the Remote Provisioning Server.
```

``mesh models rpr scan-stop``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	Stop any ongoing scanning on the Remote Provisioning Server.

``mesh models rpr link-get``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	Get the current link status of the Remote Provisioning Server.

``mesh models rpr link-close``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Close any open link on the Remote Provisioning Server.

Reviewer: ```suggestion
	Close any open links on the Remote Provisioning Server.
```

``mesh models rpr provision-remote <UUID> <NetKeyIndex> <addr>``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Provision a Mesh Node using PB-Remote

Reviewer: ```suggestion
	Provision a mesh node using the PB-Remote provisioning bearer.
```

	* ``UUID``: UUID of the unprovisioned node.
	* ``NetKeyIndex``: Network Key Index to give to the unprovisioned node.

> 	* ``addr``: Address to assign to remote device. If addr is 0, the lowest available address will be chosen.

Reviewer: ```suggestion
	* ``addr``: Address to assign to the remote device. If ``addr`` is 0, the lowest available address will be chosen.
```

``mesh models rpr reprovision-remote <addr> [<comp changed: false, true>]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 	Reprovision a Mesh Node using PB-Remote

Reviewer: ```suggestion
	Reprovision a mesh node using the PB-Remote provisioning bearer.
```

Author: Will not be better to say "the Remote Provisioning Bearer (PB-Remote)" at the end.

Reviewer: If you'd like :)
As long as you write "the remote provisioning bearer" instead (no title casing, at least not in the spec).

> 	* ``addr``: Address to assign to remote device. If addr is 0, the lowest available address will be chosen.

Reviewer: ```suggestion
	* ``addr``: Address to assign to the remote device. If ``addr`` is 0, the lowest available address will be chosen.
```

> 	* ``comp changed``: The target node has indicated that its composition data has changed. Note that the target node will reject the update if this isn't true.

Reviewer: ```suggestion
	* ``comp changed``: The Target node has indicated that its Composition Data has changed. Note that the Target node will reject the update if this isn't true.
```

	Abort the Opcodes Aggregator Sequence message. This command clears the Opcodes Aggregator Client context.

Remote Provisioning Client
--------------------------

The Remote Provisioning Client is an optional Bluetooth mesh model enabled through the :kconfig:option:`CONFIG_BT_MESH_RPR_CLI` configuration option. The Remote Provisioning Client model provides support for remote provisioning of devices into a mesh network by using the Remote Provisioning Server model.

This shell module can be used to trigger interaction between Remote Provisioning Clients and Remote Provisioning Servers on devices in a mesh network.

``mesh models rpr scan <timeout in seconds> [<UUID>]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	Start scanning for unprovisioned devices.

	* ``timeout in seconds``: Scan timeout in seconds. Must be at least 1 second.
	* ``UUID``: Device UUID to scan for. If omitted, all devices will be reported.

``mesh models rpr scan-ext <timeout in seconds> <UUID> [<AD-type> ... ]``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	Start the extended scanning for unprovisioned devices.

> 	* ``timeout in seconds``: Scan timeout in seconds. Valid values from ``@ref BT_MESH_RPR_EXT_SCAN_TIME_MIN`` to ``@ref BT_MESH_RPR_EXT_SCAN_TIME_MAX``.

Reviewer: Do we need @ref or could it be omitted?

Author: I change this to :c:macro: and that will show this in a special way in the documentation.
